.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run lint-check -- --fix
```

Comment:
⚠️ run tsc so slow the build down. We shouldn't run tsc why running eslint. I have done some tests, this is not needed
⚠️ why not calling lint instead of lint-check? this changes the previous behaviour

Reply:
Because I always forget to run lint-check. So it's always fails in the CI.
I think calling lint-check at the commit avoid many switch ^^

Reply:
Maybe, we should remove `tsc`from `lint`& `lint-check`scripts.